### PR TITLE
complete bsc example

### PR DESCRIPTION
**Pull request by ackratos**

### CODE REVIEW

**Comment by typhondev** [comment]
In the case that I use injected for binance chain wallet, how can I use both of metamask and binance chain wallet?

**Comment by ackratos** [comment]
> In the case that I use injected for binance chain wallet, how can I use both of metamask and binance chain wallet?

this is why I think my PR https://github.com/aragon/use-wallet/pull/64 should be merged so that you can declare bsc as connector... @sohkai could you please answer here?

**Comment by typhondev** [comment]
hope bsc connector PR is merged asap

File: examples/binance-chain/index.tsx [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
+      <button onClick={() => { wallet.connect('injected'); setAddress(wallet.account)}}>Connect</button>
     </div>
   )
 }
 
 render(
   <UseWalletProvider
     connectors={{
-      bsc: {
+      injected: {
```
Comment (bpierre) [inline examples/binance-chain/index.tsx:32]:
```suggestion
      bsc: {
```

I think we should use `bsc` here, to not confuse it with the native `injected` connector.

File: examples/binance-chain/index.tsx [hunk 2]
```diff
@@ -1,26 +1,37 @@
...
+
+  useEffect(() => {
+    console.log(address)
+  }, [address])
+
   return (
     <div>
       <h1>Binance Chain Connector</h1>
-      <button>Connect</button>
+      <button onClick={() => { wallet.connect('injected'); setAddress(wallet.account)}}>Connect</button>
```
Comment (bpierre) [inline examples/binance-chain/index.tsx:24]: Would you mind moving this in a separate function to improve readability?

**Comment by typhondev** [comment]
@bpierre thanks for your reply. that `bsc` is not registered in BscConnector, so I can't use that keyword.

**Comment by bpierre** [comment]
It has to do with the name used in `connect()`. If you declare it using `bsc`, then you can do `connect('bsc')` rather than `connect('injected')` − which would override another connector.

**Comment by typhondev** [comment]
that `bsc` is not registered yet because pr is not merged, so I can't use that `bsc` now.

**Comment by bpierre** [comment]
There is a difference between declaring a connector name that exists or not:

If you declare a name that doesn’t exist (e.g. `bsc`) and it looks like a connector (i.e. the object has a `web3ReactConnector()` method), then it will inject a custom connector, using the key as an identifier. This connector should then be available in `connect()`.

If you use a name that does exist, and the object doesn’t look like a connector, then it will provide a configuration for that connector.

You can find this behavior here: https://github.com/aragon/use-wallet/blob/69da02874429c048a68613af07eea558620b57db/src/connectors.js#L29-L41

**Comment by bpierre** [comment]
@typhondev merging it like this so I can make the change in the other PR.

**Comment by typhondev** [comment]
Great! thank you for your message.

**Pull request merged**